components/constellation/constellation.rs

```diff
@@ -2007,7 +2008,7 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
         // a dependent pipeline to be ready to paint.
         while let Some(valid_frame_change) = self.pending_frames.iter().rposition(|frame_change| {
             let waiting_on_dependency = frame_change.old_pipeline_id.map_or(false, |old_pipeline_id| {
-                self.pipelines.get(&old_pipeline_id).map(|pipeline| pipeline.frame).is_none()
+                self.pipelines.get(&old_pipeline_id).and_then(|pipeline| self.frames.get(&pipeline.frame_id)).is_none()
```

Comment:
I could add a `is_mature` flag to `Pipeline`, to capture the case where the frame id was `None`.